bigframes/core/blocks.py

```diff
@@ -2691,20 +2779,49 @@ def column_ids(self) -> Sequence[str]:
     def is_null(self) -> bool:
         return len(self._block._index_columns) == 0
 
+    @overload
+    def to_pandas(
+        self,
+        *,
+        ordered: Optional[bool] = ...,
+        dry_run: Literal[False] = ...,
+        allow_large_results: Optional[bool] = ...,
+    ) -> Tuple[pd.Index, Optional[bigquery.QueryJob]]:
+        ...
+
+    @overload
+    def to_pandas(
+        self,
+        *,
+        ordered: Optional[bool] = ...,
+        dry_run: Literal[True] = ...,
+        allow_large_results: Optional[bool] = ...,
+    ) -> Tuple[pd.Series, Optional[bigquery.QueryJob]]:
+        ...
+
     def to_pandas(
         self,
         *,
         ordered: Optional[bool] = None,
+        dry_run: bool = False,
         allow_large_results: Optional[bool] = None,
-    ) -> Tuple[pd.Index, Optional[bigquery.QueryJob]]:
+    ) -> Tuple[pd.Index | pd.Series, Optional[bigquery.QueryJob]]:
         """Executes deferred operations and downloads the results."""
         if len(self.column_ids) == 0:
             raise bigframes.exceptions.NullIndexError(
                 "Cannot materialize index, as this object does not have an index. Set index column(s) using set_index."
             )
         ordered = ordered if ordered is not None else True
+        if dry_run:
+            series, query_job = self._block.select_columns([]).to_pandas(
+                ordered=ordered,
+                allow_large_results=allow_large_results,
```

Comment:
I think `allow_large_results` shouldn't have an effect on dry run queries, as that controls the destination table property.